packages/@aws-cdk/aws-ecs/lib/log-drivers/aws-log-driver.ts

```diff
@@ -106,6 +130,7 @@ export class AwsLogDriver extends LogDriver {
         'awslogs-region': Stack.of(containerDefinition).region,
         'awslogs-datetime-format': this.props.datetimeFormat,
         'awslogs-multiline-pattern': this.props.multilinePattern,
+        ...(this.props.mode && { mode: this.props.mode }),
```

Comment:
Instead of using this syntax to exclude mode if it's undefined, can't we just rely on `removeEmpty` to make sure this isn't set when `mode` isn't set?
```suggestion
        'mode': this.props.mode,
```

This should have the same effect, where an undefined property results in the CF default and defining mode in the constructor will override that.

Reply:
I didn't know that CF works that way, Thanks for the suggestion

Reply:
It's more a quirk of how these `awslogs` parameters get passed down into Cloudformation and how this utility function sets their values. They all have [default values](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/using_awslogs.html#create_awslogs_logdriver_options), and they get passed into the Cloudformation for the Task Definition as an [arbitrary map](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskdefinition-containerdefinitions-logconfiguration.html). If a parameter is not specified in the CF, the default value is used. 

When we call this utility function, it returns only the keys whose values aren't `undefined`, so if `mode` isn't set, then it won't appear in the final map, and therefore CFN will assume the default value for the parameter when setting up the log configuration.